lib/dialects/abstract/query.js

```diff
@@ -514,7 +514,19 @@ AbstractQuery.prototype.isUpdateQuery = function() {
 
 AbstractQuery.prototype.handleSelectQuery = function(results) {
   var result = null;
-
+  // Map raw field names using a map of aliased fields
+  if (this.options.aliasMap) {
+    var aliasMap = this.options.aliasMap;
+    results = Utils._.map(results, function(result) {
+      return Utils._.reduce(aliasMap, function(result, alias, field) {
+        if (field in result) {
+          result[alias] = result[field];
+          delete result[field];
```

Comment:
It still passes the tests, however, it leaves remnants in the `dataValues` object.

```js
dataValues: { 
  id: 1,
  username: 'john',
  email_address: undefined,
  createdAt: Sun Jan 01 2012 02:10:10 GMT-0800 (PST),
  updatedAt: Sun Jan 01 2012 02:10:10 GMT-0800 (PST),
  emailAddress: 'john@gmail.com' 
}
```